pkg/resources/address/address.go

```diff
@@ -101,27 +101,35 @@ func (m *ModifiersBuilder) Build(ctx context.Context) ([]func(*kubermaticv1.Clus
 		subdomain = m.seed.Spec.SeedDNSOverwrite
 	}
 
-	frontProxyLoadBalancerServiceIP := ""
+	frontProxyLBServiceIP := ""
+	frontProxyLBServiceHostname := ""
 	if m.cluster.Spec.ExposeStrategy == kubermaticv1.ExposeStrategyLoadBalancer {
 		frontProxyLoadBalancerService := &corev1.Service{}
 		nn := types.NamespacedName{Namespace: m.cluster.Status.NamespaceName, Name: resources.FrontLoadBalancerServiceName}
 		if err := m.client.Get(ctx, nn, frontProxyLoadBalancerService); err != nil {
 			return nil, fmt.Errorf("failed to get the front-loadbalancer service: %v", err)
 		}
 		// Use this as default in case the implementation doesn't populate the status
-		frontProxyLoadBalancerServiceIP = frontProxyLoadBalancerService.Spec.LoadBalancerIP
+		frontProxyLBServiceIP = frontProxyLoadBalancerService.Spec.LoadBalancerIP
 		// Supposively there is only one if not..Good luck
 		for _, ingress := range frontProxyLoadBalancerService.Status.LoadBalancer.Ingress {
 			if ingress.IP != "" {
-				frontProxyLoadBalancerServiceIP = ingress.IP
+				frontProxyLBServiceIP = ingress.IP
+			}
+			if ingress.Hostname != "" {
+				frontProxyLBServiceHostname = ingress.Hostname
```

Comment:
This could be  simplified to
```
				frontProxyLBServiceIP = ingress.IP
				frontProxyLBServiceHostname = ingress.Hostname
```

Reply:
This goes back to the previous question. If we assume that there may be multiple Ingress's set, the check for emptiness is necessary to not overwrite previously retrieved value with an empty value.

Reply:
Right. 

With the checks this can happen though in case of multiple ingresses. 

ingress1: ip: nil, hostname: h1
ingress2: ip: ip2, hostname: nil 

We end up with `frontProxyLBServiceIP = ip2` and `frontProxyLBServiceHostname = h1`.

Unlikely but possible :sweat_smile:

Reply:
The thing is that I would still like to prioritize IP over hostname if we have it..
But frankly, I think we are trying to address a corner case that I have never seen :)

Reply:
BTW, I think the only really consistent solution would be to collect lists of IPs and hostnames, sort them alphabetically and then use the first one - but I am still not 100% sure if I should go that way

Reply:
I agree. Let's keep it the way it is. :+1: